### PR TITLE
Update recipe image field

### LINKED ISSUE #141: Fix image field

### LINKED ISSUE #149: Make recipe image field optional

### PR DESCRIPTION
**Pull request by jiaqi20**

### CODE REVIEW

**Review by TianYong-Goh — APPROVED** [review]
lgtm

**Review by caitlinjee — APPROVED** [review]
lgtm

File: src/main/java/seedu/address/logic/parser/ImageParser.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                byte[] buf = new byte[1024];
+                int n = 0;
+                while (-1 != (n = in.read(buf))) {
+                    out.write(buf, 0, n);
+                }
+                out.close();
+                in.close();
+                byte[] response = out.toByteArray();
+
+                //imagePath = AddRecipeCommandParser.class.getResource("/images").getPath() + "/" + filename;
```
Comment (daongochieu2810) [inline src/main/java/seedu/address/logic/parser/ImageParser.java:60]: line 57 -> 69
```//imagePath = AddRecipeCommandParser.class.getResource("/images").getPath() + "/" + filename;
                imagePath = DIRECTORY_NAME + filename;

                //imagePath = getPathsFromResourceJAR("data") + "/" + filename;
                File directory = new File(DIRECTORY_NAME);
                if (! directory.exists()){
                    directory.mkdir();
                    // If you require it to make the entire directory path including parents,
                    // use directory.mkdirs(); here instead.
                }

                FileOutputStream fos = new FileOutputStream(imagePath);
                URL jarLocation = this.getClass().getProtectionDomain().getCodeSource().getLocation();
                URL data = new URL(jarLocation, "data/" + filename);
                imagePath = data.toURI().getPath();
                fos.write(response);
                fos.close();
                //File file = FileHelp.from("file:///" + imagePath, filename);
                //imagePath = file.getPath();
                imagePath = "file://" + imagePath;```

Comment (daongochieu2810) [reply src/main/java/seedu/address/logic/parser/ImageParser.java:60]: ```DIRECTORY_NAME``` is "data/", you can declare it as static

**Review by daongochieu2810 — APPROVED** [review]

**Review by oliviajohansen — APPROVED** [review]
lgtm

**Pull request merged**